avalon/nuke/lib.py

```diff
@@ -3,82 +3,264 @@
+    except NameError as e:
+        # if it doesn't then create it
+        log.debug("Creating avalon knob: `{}`".format(e))
+        node = set_avalon_knob_data(node)
+        return get_avalon_knob_data(node)
+
+    # get data from filtered knobs
+    data = {k.replace(prefix, ''): node[k].value()
+            for k in node.knobs().keys()
+            if prefix in k}
+
+    return data
 
 
 def imprint(node, data):
     """Adding `Avalon data` into a node's Avalon Tab/Avalon knob
     also including publish knob
 
     Arguments:
-        node (list or obj): A nuke's node object either in list or individual
+        node (obj): A nuke's node object
         data (dict): Any data which needst to be imprinted
-    """
-    if not node:
-        return
-    if isinstance(node, list):
-        node = node[0]
 
-    add_avalon_tab_knob(node)
-    add_publish_knob(node)
-    node['avalon'].setValue(toml.dumps(data))
+    Returns:
+        node (obj)
+
+    Examples:
+        data = {
+            'asset': 'sq020sh0280',
+            'family': 'render',
+            'subset': 'subsetMain'
+        }
+    """
+    node = set_avalon_knob_data(node, data)
+    node = add_publish_knob(node)
 
     return node
 
 
-def add_publish_knob(node):
-    if "publish" not in node.knobs():
-        knob = nuke.Boolean_Knob("publish", "Publish")
-        knob.setFlag(0x1000)
-        knob.setValue(False)
-        node.addKnob(knob)
-    return node
+def ls_img_sequence(path):
+    """Listing all available coherent image sequence from path
 
+    Arguments:
+        path (str): A nuke's node object
 
-def fix_data_for_node_create(data):
-    for k, v in data.items():
+    Returns:
+        data (dict): with nuke formated path and frameranges
+    """
+    file = os.path.basename(path)
+    dir = os.path.dirname(path)
+    base, ext = os.path.splitext(file)
+    name, padding = os.path.splitext(base)
+
+    # populate list of files
+    files = [f for f in os.listdir(dir)
+             if name in f
+             if ext in f]
+
+    # create collection from list of files
+    collections, reminder = clique.assemble(files)
 
-        data[k] = str(v)
+    if len(collections) > 0:
+        head = collections[0].format("{head}")
+        padding = collections[0].format("{padding}") % 1
+        padding = '#' * len(padding)
+        tail = collections[0].format("{tail}")
+        file = head + padding + tail
 
-        if "True" in v:
-            data[k] = True
-        if "False" in v:
-            data[k] = False
-        if "0x" in v:
+        return {'path': os.path.join(dir, file).replace('\\', '/'),
+                'frames': collections[0].format("[{ranges}]")}
+    else:
+        return False
+
+
+def fix_data_for_node_create(data):
+    """Fixing data to be used for nuke knobs
+    """
+    for k, v in data.items():
+        if isinstance(v, six.text_type):
+            data[k] = str(v)
+        elif "0x" in str(v):
```

Comment:
Where would the `0x` value come from that it's a string? 
Anyway, should this be `str(v).startswith("0x")` instead? 

Seems like this can be more strict as opposed to it somehow messing up with a potential subset `hero_var0xtra` (worst naming convention ever). You never know what someone might name their stuff.